optimade/models/util.py

```diff
@@ -4,7 +4,17 @@
 from pydantic.validators import list_validator
 
 
-__all__ = ("NonnegativeInt", "CHEMICAL_SYMBOLS", "EXTRA_SYMBOLS", "ATOMIC_NUMBERS")
+__all__ = ("NonnegativeInt", "CHEMICAL_SYMBOLS", "EXTRA_SYMBOLS", "ATOMIC_NUMBERS", "ManualValidationError")
```

Comment:
`ManualValidationError` should be put separately in a `/common/exceptions.py` or similar I think? We should create more specific exceptions as well for the server? Or maybe just add an `exceptions.py` under `/models/`?

Reply:
Think I prefer the latter suggestion as this is quite a specific error that shouldn't really be raised anywhere else (only needs to be importable for the tests)

Reply:
Agreed.

Reply:
Better still; if you raise TypeError or ValueError inside a pydantic validator it collects all the errors and does some nicer debug output! Will remove this exception altogether.

Reply:
Same goes for `AssertException` :) Which is why we were asserting and this was also raised :)

Reply:
Ah, didn't notice... I guess this would mean the validators wouldn't run if we'd kept the assert statements though?

Reply:
Don't know what you mean exactly by "not run"? I have run into some problems with the validators already, because they worked, but because floating precision is hard :sweat_smile:

Reply:
If uvicorn spawns python with optimisations turned on, all the asserts would get skipped

Reply:
Ah - didn't know this, thanks! :)

Reply:
I've been bitten once ;)

They have a warning in the pydantic docs about it: https://pydantic-docs.helpmanual.io/usage/validators/